cla_backend/apps/knowledgebase/tests/test_csv_2_fixture.py

```diff
@@ -0,0 +1,1443 @@
+import json
+from datetime import datetime
+
+from django.test import TestCase
+from django.utils import timezone
+from django.utils.dateparse import parse_datetime
+
+from knowledgebase.management.commands._csv_2_fixture import KnowledgebaseCsvParse
+
+
+class TestCSV2Fixture(TestCase):
+    def setUp(self):
+        self.datetime_now = datetime.now().replace(tzinfo=timezone.get_current_timezone()).isoformat()
+
+    def calculate_pk_range(self, article_category_matrices):
+        min_range = min(article_category_matrices, key=lambda x: x["pk"])
+        max_range = max(article_category_matrices, key=lambda x: x["pk"])
+        return min_range["pk"], max_range["pk"]
+
+    def sort_article_category_matrices(self, output_category_matrix, expected_category_matrix):
+        output_sorted_acm = sorted(output_category_matrix, key=lambda x: x["fields"]["article_category"])
+        expected_sorted_acm = sorted(expected_category_matrix, key=lambda x: x["fields"]["article_category"])
+        return output_sorted_acm, expected_sorted_acm
+
+    def slice_records(self, output_records, expected_records, startRange=None, endRange=None):
+        output_sliced = output_records[startRange:endRange]
+        expected_sliced = expected_records[startRange:endRange]
+        return output_sliced, expected_sliced
+
+    def compare_dicts(self, outputDict, expectedDict):
+        for key, value in expectedDict.items():
+            self.assertIsInstance(value, type(outputDict[key]))
+            if isinstance(value, int):
+                self.assertEqual(outputDict[key], value)
+            elif isinstance(parse_datetime(value), datetime):
+                self.compare_datetimes(parse_datetime(outputDict[key]), parse_datetime(value))
+            else:
+                self.assertEqual(outputDict[key], value)
+
+    def compare_datetimes(self, outputDatetime, expectedDatetime):
+        time_diff = outputDatetime - expectedDatetime
+        time_diff_in_seconds = time_diff.total_seconds()
+        self.assertLessEqual(time_diff_in_seconds, 0.5)
+
+    def compare_list_of_records(self, outputList, expectedList):
+        for expectedObject, outputObject in zip(expectedList, outputList):
+            self.compare_length(outputObject, expectedObject)
+            for key, value in expectedObject.items():
+                self.assertIsInstance(value, type(outputObject[key]))
+                if isinstance(value, dict):
+                    self.compare_dicts(outputObject[key], value)
+                else:
+                    self.assertEqual(outputObject[key], value)
+
+    def compare_length(self, output, expected):
+        self.assertAlmostEqual(len(output), len(expected))
```

Comment:
Why `assertAlmostEqual`?

Reply:
Thanks for that, I forgot I was using assertEqual the whole time while reading the docs..